content/blog/2018/01/2018-01-13-jep-200.adoc

```diff
@@ -0,0 +1,104 @@
+If you are running any of the plugins listed in
+link:https://wiki.jenkins.io/display/JENKINS/Plugins+affected+by+fix+for+JEP-200[Plugins affected by fix for JEP-200],
+update them after taking the backup but before upgrading Jenkins core.
+
+If you have a way of testing the upgrade in an isolated environment before applying it to production,
+do so now.
+
+Using backups and a staging server is good advice before _any_ upgrade but especially this one,
+with a relatively high risk of regression.
+
+=== After upgrade
+
+To the extent that advance testing of the impact of this change on popular plugins has been completed,
+most users (and even plugin developers) should not notice any difference.
+If you do encounter a `java.lang.SecurityException: Rejected: some.pkg.and.ClassName` in the Jenkins UI or logs,
+you may have found a case where an unusual plugin, or an unusual usage mode of a common plugin,
+violates the existing whitelist.
+This will be visible in the Jenkins system log as a message from `jenkins.security.ClassFilterImpl` like the following:
+
+----
+some.pkg.and.ClassName in file:/var/lib/jenkins/plugins/some-plugin-name/WEB-INF/lib/some-library-1.2.jar might be dangerous, so rejecting; see https://jenkins.io/redirect/class-filter/
+----
+
+where the link would direct you here.
+
+If you find such a case, please report it in the Jenkins issue tracker, under the appropriate plugin component.
+Link it to link:https://issues.jenkins-ci.org/browse/JENKINS-47736[JENKINS-47736] and add the `JEP-200` label.
+If at all possible, include complete steps to reproduce the problem from scratch.
+Jenkins developers will strive to evaluate the reason for the violation and offer a fix in the form of a core and/or plugin update.
+For more details and current status, see
+link:https://wiki.jenkins.io/display/JENKINS/Plugins+affected+by+fix+for+JEP-200[Plugins affected by fix for JEP-200].
+
+Assuming you see no particular reason to think that the class in question has dangerous deserialization semantics, which is rare,
+it is possible to work around the problem in your own installation as a temporary expedient.
+Simply make note of any class name(s) mentioned in such log messages,
+and run Jenkins with this startup option (details will depend on your installation method):
+
+----
+-Dhudson.remoting.ClassFilter=some.pkg.and.ClassName,some.pkg.and.OtherClassName
+----
+
+== For plugin developers
+
+As a plugin developer encountering this kind of error,
+your first task is to ensure that it is reproducible in a functional (`JenkinsRule`) test
+when running
+
+[source,sh]
+----
+mvn test -Djenkins.version=2.102
+----
+
+or the next LTS version of Jenkins reported to reproduce the error (or a newer version).
+The above assumes you are using a recent 2.x parent POM.
+For certain cases you may need to use `plugin-compat-tester` to run tests against Jenkins core versions much newer than your baseline.
```

Comment:
Unhelpful without at least a link there. We're light on docs for the PCT, so not sure how much this would help.